### PR TITLE
feat(core): add ipv4 netmask function

### PR DESCRIPTION
**Pull request by allegraharris**
Adds a netmask parsing function to create additional functionality for #3523. This function takes a string argument in the format `22.59.138.9/8` and returns its netmask (`255.0.0.0`) in IPv4 format. 

**Use case:** Obtaining the broadcast bitmask for an ip address via performing bitwise NOT on the netmask. Apply a bitwise OR to this result to obtain the broadcast address of an ip address.

**Example:** 
```sql
select (~ netmask('68.11.9.2/8')) | ipv4 '68.11.9.2' -> 68.255.255.255 
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java|11|11|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        @Override
+        public Function getArg() {
+            return arg;
+        }
+
+        @Override
+        public int getIPv4(Record rec) {
+            final int val = Numbers.getIPv4Netmask(arg.getStr(rec));
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java:70]: `arg.getStr(rec)` can be null, please add a test for this case

Comment (allegraharris) [reply core/src/main/java/io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java:73]: Fixed in https://github.com/questdb/questdb/pull/3624/commits/f4a02d77c8ab120fec18cdfcca7156581a8fb029

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 13 / 13 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/IPv4StrNetmaskFunctionFactory.java|13|13|100.00%|

**Pull request merged**